File: DP/python/11055.py
n = int(input())
nums = list(map(int,input().split()))
dp = [nums[i] for i in range(n)]
dp[0] = nums[0]

for i in range(1,n):
    for j in range(i):
        if nums[i] > nums[j]:
            dp[i] = max(dp[i],dp[j]+nums[i])
print(max(dp))

File: DP/javascript/11055.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const n = Number(input[0]);
const nums = input[1].split(' ').map(Number);
const dp = nums.reduce((acc,cur)=>{
  acc.push(cur);
  return acc;
},[]);
for (let i = 1 ; i<n ; i++){
  for (let j = 0 ; j<i ; j++){
    if (nums[i] <= nums[j]) continue;
    dp[i] = Math.max(dp[i],dp[j]+nums[i]);
  }
}
console.log(Math.max(...dp))